imagetagger/imagetagger/images/views.py

```diff
@@ -119,15 +125,34 @@ def index(request):
         'active_teams': team_stats.get('active_count', 0) or 0,
         'annotation_types': annotation_types[:3],
     }
-    return TemplateResponse(
-        request, 'images/index.html', {
-            'team_creation_form': team_creation_form,
-            'imageset_creation_form': imageset_creation_form,
-            'image_sets': imagesets,
-            'userteams': userteams,
-            'stats': stats,
+
+    global_annoucements = Message.in_range(GlobalMessage.get(request.user).filter(~Q(read_by=request.user)))
```

Comment:
An index might be required to prevent the sequential scan:

```
imagetagger=# explain analyze  SELECT "tagger_messages_message"."id", "tagger_messages_message"."title", "tagger_messages_message"."content", "tagger_messages_message"."creator_id", "tagger_messages_message"."creation_time", "tagger_messages_message"."start_time", "tagger_messages_message"."expire_time", "tagger_messages_globalmessage"."message_ptr_id", "tagger_messages_globalmessage"."team_admins_only", "tagger_messages_globalmessage"."staff_only" FROM "tagger_messages_globalmessage" INNER JOIN "tagger_messages_message" ON ("tagger_messages_globalmessage"."message_ptr_id" = "tagger_messages_message"."id") WHERE (("tagger_messages_globalmessage"."team_admins_only" = false OR "tagger_messages_globalmessage"."team_admins_only" = true) AND ("tagger_messages_globalmessage"."staff_only" = false OR "tagger_messages_globalmessage"."staff_only" = true) AND NOT ("tagger_messages_globalmessage"."message_ptr_id" IN (SELECT U2."message_id" FROM "tagger_messages_message_read_by" U2 WHERE U2."user_id" = 27)) AND "tagger_messages_message"."expire_time" > '2018-11-13T00:00:00+01:00'::timestamptz AND "tagger_messages_message"."start_time" <= '2018-11-13T00:00:00+01:00'::timestamptz) ORDER BY "tagger_messages_message"."start_time" DESC LIMIT 21;
                                                                                  QUERY PLAN                                                                                   
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Limit  (cost=72.87..72.92 rows=21 width=288) (actual time=0.066..0.066 rows=0 loops=1)
   ->  Sort  (cost=72.87..73.05 rows=75 width=288) (actual time=0.063..0.063 rows=0 loops=1)
         Sort Key: tagger_messages_message.start_time DESC
         Sort Method: quicksort  Memory: 25kB
         ->  Hash Join  (cost=29.07..70.84 rows=75 width=288) (actual time=0.016..0.016 rows=0 loops=1)
               Hash Cond: (tagger_messages_globalmessage.message_ptr_id = tagger_messages_message.id)
               ->  Seq Scan on tagger_messages_globalmessage  (cost=14.81..54.81 rows=675 width=6) (never executed)
                     Filter: (((NOT team_admins_only) OR team_admins_only) AND ((NOT staff_only) OR staff_only) AND (NOT (hashed SubPlan 1)))
                     SubPlan 1
                       ->  Bitmap Heap Scan on tagger_messages_message_read_by u2  (cost=4.23..14.79 rows=10 width=4) (never executed)
                             Recheck Cond: (user_id = 27)
                             ->  Bitmap Index Scan on tagger_messages_message_read_by_user_id_fb22e6ae  (cost=0.00..4.23 rows=10 width=0) (never executed)
                                   Index Cond: (user_id = 27)
               ->  Hash  (cost=13.90..13.90 rows=29 width=282) (actual time=0.006..0.007 rows=0 loops=1)
                     Buckets: 1024  Batches: 1  Memory Usage: 8kB
                     ->  Seq Scan on tagger_messages_message  (cost=0.00..13.90 rows=29 width=282) (actual time=0.006..0.006 rows=0 loops=1)
                           Filter: ((expire_time > '2018-11-13 00:00:00+01'::timestamp with time zone) AND (start_time <= '2018-11-13 00:00:00+01'::timestamp with time zone))
 Planning time: 0.781 ms
 Execution time: 0.180 ms
(19 rows)
```

Reply:
The current solution shows no problems in extensive testing.